alns/accept/LateAcceptanceHillClimbing.py

```diff
@@ -0,0 +1,78 @@
+from collections import deque
+
+from alns.accept.AcceptanceCriterion import AcceptanceCriterion
+
+
+class LateAcceptanceHillClimbing(AcceptanceCriterion):
+    """
+    The Late Acceptance Hill Climbing (LAHC) criterion accepts a candidate
+    solution when it is better than the current solution from a number of
+    iterations ago.
+
+    This implementation is based on the description of LAHC in [1].
+
+    Parameters
+    ----------
+    lookback_period: int
+        Nonnegative integer specifying which solution to compare against
+        for late acceptance. In particular, LAHC compares against the
+        then-current solution from `lookback_period` iterations ago.
+        If set to 0, then LAHC reverts to regular hill climbing.
+    greedy: bool
+        If set, LAHC always accepts a candidate that is better than the
+        current solution.
+    better_history: bool
+        If set, LAHC uses a history management strategy where current solutions
+        are stored only if they improve the then-current solution from
+        `lookback_period` iterations ago; otherwise the then-current solution
+        is stored again.
+
+    References
+    ----------
+    [1]: Burke, E. K., & Bykov, Y. The late acceptance hill-climbing heuristic.
+         *European Journal of Operational Research* (2017), 258(1), 70-78.
+    """
+
+    def __init__(
+        self,
+        lookback_period: int,
+        greedy: bool = False,
+        better_history: bool = False,
+    ):
+        self._lookback_period = lookback_period
+        self._greedy = greedy
+        self._better_history = better_history
+
+        if not isinstance(lookback_period, int) or lookback_period < 0:
+            raise ValueError("lookback_period must be a non-negative integer.")
```

Comment:
Refactor: no need to check the type. If it's not something that looks like an integer, `deque` will complain in the next statement. 

We already indicate the argument should be integer-like via the type hint. I think that's enough.